packages/trader/src/Modules/Trading/Components/Form/Purchase/cancel-deal-info.tsx

```diff
@@ -5,19 +5,23 @@ import { isDesktop, isMobile, getDecimalPlaces } from '@deriv/shared';
 import { localize } from '@deriv/translations';
 import { observer } from '@deriv/stores';
 import { useTraderStore } from 'Stores/useTraderStores';
+import { TProposalTypeInfo } from 'Types';
 
-const CancelDealInfo = observer(({ proposal_info }) => {
+const CancelDealInfo = observer(({ proposal_info }: { proposal_info: TProposalTypeInfo }) => {
     const { currency, has_cancellation } = useTraderStore();
     const { id, cancellation, has_error } = proposal_info;
     const error = has_error || !id;
     const [is_row_layout, setIsRowLayout] = React.useState(false);
 
-    const ref = React.useRef(null);
+    const ref = React.useRef<HTMLDivElement>(null);
 
     React.useEffect(() => {
         if (ref.current) {
             const el_height = ref.current.parentElement?.clientHeight;
-            if ((el_height > 21 && isDesktop()) || ((el_height > 21 || getDecimalPlaces(currency) > 2) && isMobile())) {
+            if (
+                el_height &&
```

Comment:
@kate-deriv unfortunately, this condition is broken. now when `el_height` is falsy, it returns its value instead of `false` for desktop and instead of `true` for mobile.

Reply:
Will correct, thank you